### PR TITLE
[proposal] Add proposal for NMS E2E/Regression Testing

### PR DESCRIPTION
**Pull request by andreilee**
Signed-off-by: Andrei Lee <andreilee@fb.com>

## Summary

Adds a proposal for NMS e2e/regression testing via integration of Testim. For feedback and discussion, please direct all comments to https://github.com/magma/magma/discussions/6077

## Test Plan

N/A

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: docs/readmes/proposals/p014_nms_regression_testing.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+5. Create/update/delete a subscriber
+6. Create/update/delete an APN
+7. Create/update/delete a policy
+8. Create/update/delete a traffic profile
+9. Create/update/delete a rating group
+10. Search through NMS metrics explorer
+11. Add/update/delete an alert receiver
+12. Add/update/delete an alert rule
+13. Create/update/delete an NMS user
+
```
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Verify addition of organization and associated users

Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Verify LTE networks dashboard

File: docs/readmes/proposals/p014_nms_regression_testing.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
```
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Verify all charting components

File: docs/readmes/proposals/p014_nms_regression_testing.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+## Implementation
```
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Verify logs/events

File: docs/readmes/proposals/p014_nms_regression_testing.md (hunk 1 above)
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Verify grafana dashboard

File: docs/readmes/proposals/p014_nms_regression_testing.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+Testim, for at least one year.
+
+**Tests**
+
+The following tests are proposed to verify pure-play FWA usage of the Magma
+NMS:
+
+1. Create/update/delete an LTE network
+2. Create/update/delete an LTE AGW
+3. Create and register an eNodeB configuration
```
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Variants - managed/unmanaged eNodeB configuration

File: docs/readmes/proposals/p014_nms_regression_testing.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+Facebook will pay for the associated costs with using the service provided by
+Testim, for at least one year.
+
+**Tests**
+
+The following tests are proposed to verify pure-play FWA usage of the Magma
+NMS:
+
+1. Create/update/delete an LTE network
+2. Create/update/delete an LTE AGW
```
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Variants to be tested here - gateways with different IP allocation schemes including IP_POOL, bridged configuration and STATIC_IP configuration

File: docs/readmes/proposals/p014_nms_regression_testing.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+Facebook will pay for the associated costs with using the service provided by
+Testim, for at least one year.
+
+**Tests**
+
+The following tests are proposed to verify pure-play FWA usage of the Magma
+NMS:
+
+1. Create/update/delete an LTE network
```
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Add tests around verifying plmn restriction, dataplans, imei restriction etc

File: docs/readmes/proposals/p014_nms_regression_testing.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+2. Create/update/delete an LTE AGW
+3. Create and register an eNodeB configuration
+4. Create/update/delete a gateway pool and register the LTE AGW
+5. Create/update/delete a subscriber
```
Comment (karthiksubraveti) [inline docs/readmes/proposals/p014_nms_regression_testing.md]: Verify bulk subscriber add

**Comment by hcgatewood** [comment]
👍

**Comment by electronjoe** [comment]
Thank you!

**Review by electronjoe — APPROVED** [review]

**Pull request merged**